src/editor/components/components/AddLayerPanel/createLayerFunctions.js

```diff
@@ -149,11 +149,22 @@ function createPrimitiveGeometry() {
   Events.emit('entitycreated', newEl);
 }
 
+function createIntersection() {
+  const newEl = document.createElement('a-entity');
+  newEl.setAttribute('intersection', '');
+  newEl.setAttribute('data-layer-name', 'Street intersection');
+  const parentEl = document.querySelector('#street-container');
+  parentEl.appendChild(newEl);
+  // update sceneGraph
+  Events.emit('entitycreated', newEl);
+}
+
```

Comment:
To properly implement undo/redo, we need to stop using such code to create an entity. We should use the proper api 
```js
Events.emit('entitycreate', {
  'data-layer-name': 'Street intersection',
  components: {intersection: ''}
});
```
but we need the changes for createEntity from
https://github.com/3DStreet/3dstreet/pull/537/files#diff-2f5a94d2538a12f206cf431af70f02620687da9738381f7be44feeaf809bc5b3R618
We can do that in another PR and modify all the entity creation here.

Reply:
I used `emit('entitycrated')` here to update the scene graph after adding a new element, not to create the entity. But I know that now there is an `updatescenegraph` event, so I'll change to it.

Reply:
No no don't change it. entitycreated here is better than updatescenegraph. But we'll need to rewrite all those create functions later that's what I'm saying.

Reply:
Ok, got it